### PR TITLE
Multivariate scenario

### PR DESCRIPTION
**Pull request by bradstallion**
Hi Emanuel,
thanks for your code.

This branch extends it to multivariate series, where features are (slot, signal).
It should be compatible with your scenario (univariate).
Moreover, it includes:

- updated requirements
- setup.py includes requirements.txt
- some logging
- the test demo/testmultivariate.py (requires sktime)

I hope you'll find this changes useful... and correct! :-)

Please feel free to test and pull it, if you want.
Thanks
Bye

### CODE REVIEW

File: lime_timeseries.py [hunk 1]
```diff
@@ -143,40 +176,88 @@ def distance_fn(x):
...
         for i, num_inactive in enumerate(deact_per_slice, start=1):
+            logging.info("sample %d, inactivating %d", i, num_inactive)
             # choose random slices indexes to deactivate
             inactive_idxs = np.random.choice(features_range, num_inactive,
                                              replace=False)
-            perturbation_matrix[i, inactive_idxs] = 0
+            num_channels_to_perturb = np.random.randint(1, num_channels+1)
+            tmp_channels = list(range(num_channels))
+            np.random.shuffle(tmp_channels)
+            channels_to_perturb = tmp_channels[:num_channels_to_perturb]
```
Comment (emanuel-metzenthin) [inline lime_timeseries.py]: I think this could be done more elegantly using np.choice(tmp_channels, num_channels_toperturb, replace=False)

File: lime_timeseries.py [hunk 2]
```diff
@@ -143,40 +176,88 @@ def distance_fn(x):
...
                 x, x[0].reshape([1, -1]), metric='cosine').ravel() * 100
 
-        values_per_slice = math.ceil(len(timeseries) / num_slices)
+        num_channels = 1
+        len_ts = len(timeseries)
+        if len(timeseries.shape) > 1:  # multivariate
+            num_channels, len_ts = timeseries.shape
+        
+        values_per_slice = math.ceil(len_ts / num_slices)
         deact_per_slice = np.random.randint(1, num_slices + 1, num_samples - 1)
```
Comment (emanuel-metzenthin) [inline lime_timeseries.py]: not part of this change, but this should be named deact_per_sample

File: lime_timeseries.py [hunk 3]
```diff
@@ -3,7 +3,33 @@
 from lime import explanation
 from lime import lime_base
 import math
+import logging
 
+class MultiTSDomainMapper(explanation.DomainMapper):
```
Comment (emanuel-metzenthin) [inline lime_timeseries.py]: Maybe TSDomainMapper or TimeSeriesDomainMapper? Because it also handles univariate TS now.

**Review by emanuel-metzenthin — CHANGES_REQUESTED** [review]

File: demo/testmultivariate.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+import matplotlib.pyplot as plt
+
+from lime import explanation
+from lime import lime_base
+import lime_timeseries as lime_ts
+
+# random dataset
+def genDataset(num_samples, num_channels, num_points):
+    X = np.random.random((num_samples, num_channels, num_points))
```
Comment (emanuel-metzenthin) [inline demo/testmultivariate.py:21]: Set initial random seed, to avoid non-deterministic test?

File: lime_timeseries.py (hunk 3 above)
Comment (bradstallion) [reply lime_timeseries.py]: commit e60b38b2cab3e529af89869ff183a077c5f1a1da

File: lime_timeseries.py (hunk 2 above)
Comment (bradstallion) [reply lime_timeseries.py]: @emanuel-metzenthin should I change it?

File: demo/testmultivariate.py (hunk 4 above)
Comment (bradstallion) [reply demo/testmultivariate.py:21]: commit f6686aa44e9e902f14fc6a0fe3ef184d6d49a124

File: lime_timeseries.py (hunk 1 above)
Comment (bradstallion) [reply lime_timeseries.py]: commit 25b26816ef1cb1a69432f93b90778b03cad9b7c3

File: lime_timeseries.py (hunk 2 above)
Comment (bradstallion) [reply lime_timeseries.py]: I did it in c5da3190d2912257b7f23ad82a35ef35c1098417

**Review by emanuel-metzenthin — APPROVED** [review]
Thanks a lot for your contribution!
